src/aap_eda/api/serializers/activation.py

```diff
@@ -200,8 +239,35 @@ def create(self, validated_data):
         validated_data["rulebook_rulesets"] = rulebook.rulesets
         validated_data["git_hash"] = rulebook.project.git_hash
         validated_data["project_id"] = rulebook.project.id
+        if validated_data.get("event_streams"):
+            validated_data["rulebook_rulesets"] = self._updated_ruleset(
+                validated_data
+            )
         return super().create(validated_data)
 
+    def _updated_ruleset(self, validated_data):
```

Comment:
@hsong-rh self is not being used can we make it a static function and just call it.

Reply:
Making this a static function solely on `self` not being referenced is not a sound basis; `self` is only visible to the implementation as that is how `python` implemented object handling.
If there is a case where it is known that this will be reused then it's reasonable.

Reply:
I should add that making it a static method of the class rather than an external function would be a preferable approach, barring the need for reuse.